webapp/javascript/services/users.ts

```diff
@@ -0,0 +1,26 @@
+import { Result } from '@utils/fp';
+import { Users, parse } from '@models/users';
+import type { ZodError } from 'zod';
+import { request } from './base';
+import type { RequestError } from './base';
+
+export interface FetchUsersError {
+  message?: string;
+}
+
+export async function fetchUsers(): Promise<
+  Result<Users, RequestError | ZodError>
+> {
+  const response = await request('/api/users');
+
+  try {
+    if (response.isOk) {
+      console.log('Parsing', response.value);
+      return parse(response.value);
+    }
+  } catch (e) {
+    console.error(e);
+  }
```

Comment:
Sorry, why was this needed? The whole point of using `Result` is to not deal with exceptions, so why was this throwing an exception? Let me know, maybe we messed something up in the base implementation.

Reply:
it may throw an exception once it doesn't pass `zod` validation inside `parse(response.value);`